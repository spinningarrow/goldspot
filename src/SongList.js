import React from 'react'
import SongItem from './SongItem'

const SongList = ({ heading, items = [] }) =>
	<div className="song-list">
		<h1>{heading}</h1>
		<ul>{
			items.map(({ id, albumArt, title, url }) => <SongItem
				key={id}
				albumArt={albumArt}
				title={title}
				url={url}
			/>)
		}</ul>
	</div>

export default SongList
