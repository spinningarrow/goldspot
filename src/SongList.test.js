import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import SongList from './SongList'

const renderer = new ShallowRenderer()

const items = [
	{
		id: 1,
		albumArt: 'art1.jpg',
		title: 'The Best Song',
		artist: 'Artist Five',
		url: 'https://the/best/song',
	},
	{
		id: 2,
		albumArt: 'art2.jpg',
		title: 'Song #2',
		artist: 'Arty',
		url: 'https://the/second/song',
	},
]

test('it renders a list of SongItems', () => {
	const tree = renderer.render(<SongList heading="Some Songs" items={items} />)
	expect(tree).toMatchSnapshot()
})
