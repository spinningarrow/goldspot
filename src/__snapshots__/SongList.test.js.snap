// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a list of SongItems 1`] = `
<div
  className="song-list"
>
  <h1>
    Some Songs
  </h1>
  <ul>
    <SongItem
      albumArt="art1.jpg"
      artist="Artist Five"
      title="The Best Song"
      url="https://the/best/song"
    />
    <SongItem
      albumArt="art2.jpg"
      artist="Arty"
      title="Song #2"
      url="https://the/second/song"
    />
  </ul>
</div>
`;
